src/requests.jl

```diff
@@ -92,9 +127,32 @@ end
                              path::AbstractString;
                              body::Union{AbstractString, Nothing} = nothing,
                              headers::AbstractDict = Dict{String, String}(),
-                             query::Union{AbstractDict, Nothing} = nothing)
-    full_url = _generate_full_url(client,
-                                  path)
+                             query::Union{AbstractDict, Nothing} = nothing,
+                             require_base_url::Symbol = :strict)
+    # Check that `require_base_url` is valid
+    if require_base_url !== :strict && require_base_url !== :host && require_base_url !== :scheme && require_base_url !== :no
+        throw(ArgumentError("The provided keyword argument `require_base_url = $(require_base_url)` is invalid: `require_base_url` must be `:full`, `:host`, `:scheme`, or `:no`."))
```

Comment:
These two lines are inconsistent. Line 133 has the options of strict/host/scheme/no. Line 134 has the options of full/host/scheme/no.

Reply:
Ah, I used `full` in a draft.

Reply:
```suggestion
    if require_base_url !== :strict && require_base_url !== :host && require_base_url !== :scheme && require_base_url !== :no
        throw(ArgumentError("The provided keyword argument `require_base_url = $(require_base_url)` is invalid: `require_base_url` must be `:strict`, `:host`, `:scheme`, or `:no`."))
```